mlir/lib/Dialect/Bufferization/Transforms/BufferDeallocationSimplification.cpp

```diff
@@ -77,18 +77,12 @@ static bool distinctAllocAndBlockArgument(Value v1, Value v2) {
   return areDistinct(v1Base, v2Base) || areDistinct(v2Base, v1Base);
 }
 
-/// Checks if `memref` may or must alias a MemRef in `otherList`. It is often a
-/// requirement of optimization patterns that there cannot be any aliasing
-/// memref in order to perform the desired simplification. The `allowSelfAlias`
-/// argument indicates whether `memref` may be present in `otherList` which
-/// makes this helper function applicable to situations where we already know
-/// that `memref` is in the list but also when we don't want it in the list.
+/// Checks if `memref` may potentially alias a MemRef in `otherList`. It is
```

Comment:
This was already there, but I find the "It is often a requirement ..." part a bit awkwardly phrased. 
Can we rephrase this into something like

This test if often needed by optimizations that are only valid when aliasing cannot occur

or something like that (feel free to correct my English please ;-)

Reply:
I'm going to rewrite this entire function in a future commit that I haven't sent out for review yet. Another issue here is that we are interested in "same allocation" and not "aliasing". These are slightly different properties. I'm going to update that part of the comment as well.